File: benchmarks/fair/fib.go
package main
import ("fmt";"os";"strconv")
func fib(n int64) int64 { if n<=1 { return n }; return fib(n-1)+fib(n-2) }
func main(){
    n:=int64(32)
    if v,err:=strconv.ParseInt(os.Getenv("BENCH_N"),10,64); err==nil&&v>0 { n=v }
    fmt.Println(fib(n))
}

File: benchmarks/fair/fib.java
public class fib{
  static long f(long n){ return n<=1?n:f(n-1)+f(n-2); }
  public static void main(String[] a){
    long n=32; String e=System.getenv("BENCH_N");
    if(e!=null){ try{ long v=Long.parseLong(e); if(v>0)n=v; }catch(Exception x){} }
    System.out.println(f(n));
  }
}
